BDArmory/Control/MissileFire.cs

```diff
@@ -8190,6 +8172,28 @@ public bool GetLaunchAuthorization(Vessel targetV, MissileFire mf, MissileBase m
             return launchAuthorized;
         }
 
+        /// <summary>
+        /// Check if missile is currently unable to be guided, does not apply for intentionally unguided missiles
+        /// </summary>
+        /// <returns>true if missile is unguided</returns>
+        public bool UnguidedMissile(MissileBase ml, float distanceToTarget)
+        {
+            bool unguidedWeapon = false;
+            MissileLauncher mlauncher = ml as MissileLauncher;
+
+            unguidedWeapon = ml.GuidanceMode == MissileBase.GuidanceModes.None ||
+                ml.TargetingMode switch
+                {
+                    MissileBase.TargetingModes.Laser => BDATargetManager.ActiveLasers.Count <= 0,
+                    MissileBase.TargetingModes.Radar => !_sonarsEnabled && (!ml.radarLOAL || (mlauncher != null && ml.radarTimeout < ((distanceToTarget - ml.activeRadarRange) / mlauncher.optimumAirspeed))),
```

Comment:
Since this is now unified this now needs to be ``MissileBase.TargetingModes.Radar => (mlauncher .torpedo ? _sonarsEnabled : _radarsEnabled)...`` and ``MissileBase.TargetingModes.Inertial => !((mlauncher .torpedo ? _sonarsEnabled : _radarsEnabled)|| _irstsEnabled),`` to set the proper radar subType it needs to check for.